Travis: unit test timeout flake

https://travis-ci.org/GoogleContainerTools/skaffold/jobs/383383678

```
=== RUN   TestDockerContext
2018/05/24 20:23:34 Unable to read "/Users/travis/.docker/config.json", falling back on anonymous: open /Users/travis/.docker/config.json: no such file or directory
coverage: 30.1% of statements
panic: test timed out after 1m0s
goroutine 31 [running]:
testing.(*M).startAlarm.func1()
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:1240 +0xfc
created by time.goFunc
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/time/sleep.go:172 +0x44
goroutine 1 [chan receive]:
testing.(*T).Run(0xc4203625a0, 0x1e5c6d0, 0x11, 0x1ed9ee8, 0x108a701)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc4203620f0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc4203620f0, 0xc4204cfd40)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:777 +0xd0
testing.runTests(0xc4203a6a40, 0x270e2e0, 0x9, 0x9, 0x20)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc420332300, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:978 +0x171
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.TestMain(0xc420332300)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/image_test.go:34 +0xb0
main.main()
	_testmain.go:110 +0x208
goroutine 19 [syscall, 1 minutes]:
os/signal.signal_recv(0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/runtime/sigqueue.go:139 +0xa7
os/signal.loop()
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/os/signal/signal_unix.go:28 +0x41
goroutine 20 [chan receive]:
github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x2725b80)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog/glog.go:882 +0x8b
created by github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog.init.0
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog/glog.go:410 +0x203
goroutine 41 [select]:
io.(*pipe).Read(0xc42069ce10, 0xc4200e4928, 0x200, 0x200, 0xc420657c60, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/io/pipe.go:50 +0x115
io.(*PipeReader).Read(0xc4200c60c0, 0xc4200e4928, 0x200, 0x200, 0x1d05f78, 0xc420073a28, 0x1071c8a)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/io/pipe.go:127 +0x4c
io.ReadAtLeast(0x1f556e0, 0xc4200c60c0, 0xc4200e4928, 0x200, 0x200, 0x200, 0x1071bab, 0x1f56e40, 0x2744288)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/io/io.go:309 +0x86
io.ReadFull(0x1f556e0, 0xc4200c60c0, 0xc4200e4928, 0x200, 0x200, 0x1f56e40, 0x2744288, 0x1f556e0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/io/io.go:327 +0x58
archive/tar.(*Reader).readHeader(0xc4200e4900, 0xc4200c60c0, 0xc4200e4928, 0x0, 0x200)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/archive/tar/reader.go:342 +0x71
archive/tar.(*Reader).next(0xc4200e4900, 0x240, 0x1d8e1c0, 0x1)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/archive/tar/reader.go:78 +0x145
archive/tar.(*Reader).Next(0xc4200e4900, 0xc4200e4900, 0xc4200c60d0, 0xc42001cc80)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/archive/tar/reader.go:52 +0x4e
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.TestDockerContext(0xc4203625a0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/context_test.go:55 +0x8d7
testing.tRunner(0xc4203625a0, 0x1ed9ee8)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/testing/testing.go:824 +0x2e0
goroutine 42 [select]:
net/http.(*Transport).getConn(0x270e680, 0xc420090c00, 0x0, 0xc4202c2a00, 0x5, 0xc4202c2a40, 0x13, 0x0, 0x0, 0x1046d86)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/transport.go:962 +0x558
net/http.(*Transport).RoundTrip(0x270e680, 0xc4204ea500, 0x270e680, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/transport.go:409 +0x632
net/http.send(0xc4204ea500, 0x1f559c0, 0x270e680, 0x0, 0x0, 0x0, 0xc42000e0b8, 0xf8, 0xc4204a7328, 0x1)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/client.go:252 +0x185
net/http.(*Client).send(0xc420090a80, 0xc4204ea500, 0x0, 0x0, 0x0, 0xc42000e0b8, 0x0, 0x1, 0x1012c18)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/client.go:176 +0xfa
net/http.(*Client).Do(0xc420090a80, 0xc4204ea500, 0xc4202c2a00, 0x1b, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/client.go:615 +0x28d
net/http.(*Client).Get(0xc420090a80, 0xc4202c2a00, 0x1b, 0x2, 0x2, 0xc4202c2a00)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/client.go:396 +0x9d
github.com/GoogleContainerTools/skaffold/vendor/github.com/google/go-containerregistry/v1/remote/transport.ping(0x0, 0x0, 0x1f559c0, 0x270e680, 0x0, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/google/go-containerregistry/v1/remote/transport/ping.go:63 +0x203
github.com/GoogleContainerTools/skaffold/vendor/github.com/google/go-containerregistry/v1/remote/transport.New(0x0, 0x0, 0x1f53fe0, 0x2743ce0, 0x1f559c0, 0x270e680, 0xc420604200, 0x1, 0x1, 0xc4205f4095, ...)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/google/go-containerregistry/v1/remote/transport/transport.go:47 +0x67
github.com/GoogleContainerTools/skaffold/vendor/github.com/google/go-containerregistry/v1/remote.Image(0x1f72f80, 0xc420090a20, 0x1f53fe0, 0x2743ce0, 0x1f559c0, 0x270e680, 0x0, 0x0, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/google/go-containerregistry/v1/remote/image.go:50 +0x139
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.remoteImage(0xc4205f4095, 0x6, 0x0, 0x0, 0x1f568c0, 0xc4206041a0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/image.go:175 +0x158
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.retrieveRemoteConfig(0xc4205f4095, 0x6, 0xc4205f4095, 0x6, 0x0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/parse.go:289 +0x43
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.retrieveImage(0xc4205f4095, 0x6, 0xc4205f4095, 0x6, 0x1)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/parse.go:268 +0x199
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.processBaseImage(0xc4204ee770, 0xc42000e008, 0xc4200b8700, 0x0, 0x0, 0x6369666974726543)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/parse.go:240 +0x13b
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.readDockerfile(0xc42001cc80, 0x42, 0x1e54117, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/parse.go:78 +0x2cb
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.GetDockerfileDependencies(0x1e54117, 0xa, 0xc42001cc80, 0x42, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/parse.go:118 +0x93
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.CreateDockerTarContext(0x1f55700, 0xc4200c60d0, 0x1e54117, 0xa, 0xc42001cc80, 0x42, 0xc4203625a0, 0x7bd6fc5f7)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/context.go:29 +0x5a
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.TestDockerContext.func1(0xc4200c60d0, 0xc42001cc80, 0x42)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/context_test.go:44 +0x64
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.TestDockerContext
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/context_test.go:43 +0x765
goroutine 28 [select]:
net.(*Resolver).LookupIPAddr(0x2723950, 0x1f6e440, 0xc42028f7a0, 0xc4202c2a40, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/lookup.go:212 +0x50d
net.(*Resolver).internetAddrList(0x2723950, 0x1f6e440, 0xc42028f7a0, 0x1e4cdb0, 0x3, 0xc4202c2a40, 0x13, 0x0, 0x0, 0x0, ...)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/ipsock.go:293 +0x5c4
net.(*Resolver).resolveAddrList(0x2723950, 0x1f6e440, 0xc42028f7a0, 0x1e4d4a2, 0x4, 0x1e4cdb0, 0x3, 0xc4202c2a40, 0x13, 0x0, ...)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/dial.go:193 +0x50c
net.(*Dialer).DialContext(0xc4200cc300, 0x1f6e400, 0xc4200b6048, 0x1e4cdb0, 0x3, 0xc4202c2a40, 0x13, 0x0, 0x0, 0x0, ...)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/dial.go:375 +0x22b
net.(*Dialer).DialContext-fm(0x1f6e400, 0xc4200b6048, 0x1e4cdb0, 0x3, 0xc4202c2a40, 0x13, 0x10051f6, 0x12b7339, 0xc4200b6048, 0x1ce0b40)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/transport.go:46 +0x73
net/http.(*Transport).dial(0x270e680, 0x1f6e400, 0xc4200b6048, 0x1e4cdb0, 0x3, 0xc4202c2a40, 0x13, 0x0, 0x0, 0x0, ...)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/transport.go:898 +0x20d
net/http.(*Transport).dialConn(0x270e680, 0x1f6e400, 0xc4200b6048, 0x0, 0xc4202c2a00, 0x5, 0xc4202c2a40, 0x13, 0x0, 0x0, ...)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/transport.go:1143 +0x317
net/http.(*Transport).getConn.func4(0x270e680, 0x1f6e400, 0xc4200b6048, 0xc420090c30, 0xc42009ca80)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/transport.go:957 +0x78
created by net/http.(*Transport).getConn
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/http/transport.go:956 +0x363
goroutine 29 [select]:
net.cgoLookupIP(0x1f6e3c0, 0xc4200b8840, 0xc4202c2a40, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/cgo_unix.go:212 +0x19f
net.(*Resolver).lookupIP(0x2723950, 0x1f6e3c0, 0xc4200b8840, 0xc4202c2a40, 0xf, 0x0, 0xc4200a1b00, 0xc4200b8000, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/lookup_unix.go:95 +0x12d
net.(*Resolver).(net.lookupIP)-fm(0x1f6e3c0, 0xc4200b8840, 0xc4202c2a40, 0xf, 0x102b629, 0x8, 0xc4200b8000, 0x0, 0xc420619ea0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/lookup.go:192 +0x56
net.glob..func10(0x1f6e3c0, 0xc4200b8840, 0xc4206042a0, 0xc4202c2a40, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/hook.go:19 +0x52
net.(*Resolver).LookupIPAddr.func1(0x0, 0x0, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/lookup.go:206 +0xd8
internal/singleflight.(*Group).doCall(0x2723940, 0xc4200140f0, 0xc4202c2a40, 0xf, 0xc420090d20)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/internal/singleflight/singleflight.go:95 +0x2e
created by internal/singleflight.(*Group).DoChan
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/internal/singleflight/singleflight.go:88 +0x2d0
goroutine 30 [syscall]:
net._C2func_getaddrinfo(0xc4205f4390, 0x0, 0xc420090db0, 0xc42000e0c8, 0x0, 0x0, 0x0)
	_cgo_gotypes.go:92 +0x55
net.cgoLookupIPCNAME.func1(0xc4205f4390, 0x0, 0xc420090db0, 0xc42000e0c8, 0x10, 0x10, 0xc4200b8000)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/cgo_unix.go:149 +0x13b
net.cgoLookupIPCNAME(0xc4202c2a40, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/cgo_unix.go:149 +0x174
net.cgoIPLookup(0xc42028f920, 0xc4202c2a40, 0xf)
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/cgo_unix.go:201 +0x4d
created by net.cgoLookupIP
	/Users/travis/.gimme/versions/go1.10.1.darwin.amd64/src/net/cgo_unix.go:211 +0xaf
FAIL	github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker	60.050s
```

--- a/pkg/skaffold/docker/context_test.go
+++ b/pkg/skaffold/docker/context_test.go
@@ -31,6 +31,11 @@ func TestDockerContext(t *testing.T) {
 	tmpDir, cleanup := testutil.TempDir(t)
 	defer cleanup()
 
+	RetrieveImage = mockRetrieveImage
+	defer func() {
+		RetrieveImage = retrieveImage
+	}()
+
 	os.Mkdir(filepath.Join(tmpDir, "files"), 0750)
 	ioutil.WriteFile(filepath.Join(tmpDir, "files", "ignored.txt"), []byte(""), 0644)
 	ioutil.WriteFile(filepath.Join(tmpDir, "files", "included.txt"), []byte(""), 0644)

